app/Api/Internal/Controllers/AchievementController.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Api\Internal\Controllers;
+
+use App\Api\Internal\Requests\DemoteAchievementRequest;
+use App\Community\Enums\ArticleType;
+use App\Http\Controller;
+use App\Models\Achievement;
+use App\Models\User;
+use App\Platform\Enums\AchievementFlag;
+use App\Platform\Services\GameTopAchieversService;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\DB;
+
+class AchievementController extends Controller
+{
+    /**
+     * Demote an achievement from Official/Core/Published to Unofficial/Unpublished status.
+     *
+     * This endpoint allows authorized service accounts to demote achievements,
+     * optionally updating the achievement title in the process (eg: prefixing
+     * with "DEMOTED AS UNWELCOME CONCEPT").
+     *
+     * @example Request payload:
+     * ```json
+     * {
+     *   "data": {
+     *     "type": "achievement-demotion",
+     *     "attributes": {
+     *       "achievementId": 123,
+     *       "username": "DevCompliance",
+     *       "title": "DEMOTED AS UNWELCOME CONCEPT - Original Title" // optional
+     *     }
+     *   }
+     * }
+     * ```
```

Comment:
If this is supposed to be [JSON-API compliant](https://jsonapi.org/format/#crud-updating), I think it needs to be:
```
"data": {
  "type": "achievement",
  "id": 123,
  "attributes": {
    "flags": 5,
    "actingUser": "DevCompliance",
    "title": "DEMOTED AS UNWELCOME CONCEPT - Original Title" // optional
  }
}
```
And the URL should be `http://localhost:64000/api/internal/achievements/123`.

I don't see anything about indirect mutations in the JSON-API document, but maybe we could treat it as a relationship? `http://localhost:64000/api/internal/achievements/123/relationships/demote`

I guess it just depends on how strictly we want to follow the JSON-API spec.